docs/tutorials/versioning-schemas.md

```diff
@@ -1,16 +1,209 @@
 # Versioning Schemas
 
+## Overview
 
-During development, it is natural that the fields on objects in OTIO change.  To accommodate this, OTIO has a system for handling version differences and upgrading older schemas to new ones.  There are two components:
+This document describes OpenTimelineIO's systems for dealing with different schema versions when reading files, writing files, or during development of the library itself.  It is intended for developers who are integrating OpenTimelineIO into their pipelines or applications, or working directly on OpenTimelineIO.
 
-1. `serializeable_label` on the class has a name and version field: `Foo.5` -- `Foo` is the schema name and `5` is the version.
-2. `upgrade_function_for` decorator
+TL;DR for users: OpenTimelineIO should be able to read files produced by older versions of the library and be able to write files that are compatible with older versions of the library from newer versions.
 
+## Schema/Version Introduction
 
-Changing a Field
----------------------
+Each SerializeableObject (the base class of OpenTimelineIO) has `schema_name` and `schema_version` fields.  The `schema_name` is a string naming the schema, for example, `Clip`, and the `schema_version` is an integer of the current version number, for example, `3`.
 
-For example, lets say you have class:
+SerializeableObjects can be queried for these using the `.schema_name()` and `.schema_version()` methods.  For a given release of the OpenTimelineIO library, in-memory objects the library creates will always be the same schema version.  In other words, if `otio.schema.Clip()` instantiates an object with `schema_version` 2, there is no way to get an in-memory `Clip` object with version 1.
+
+OpenTimelineIO can still interoperate with older and newer versions of the library by way of the schema upgrading/downgrading system.  As OpenTimelineIO deserializes json from a string or disk, it will upgrade the schemas to the version supported by the library before instantiating the concrete in-memory object.  Similarly, when serializing OpenTimelineIO back to disk, the user can instruct OpenTimelineIO to downgrade the JSON to older versions of the schemas.  In this way, a newer version of OpenTimelineIO can read files with older schemas, and a newer version of OpenTimelineIO can generate JSON with older schemas in it.
+
+## Schema Upgrading
+
+Once a type is registerd to OpenTimelineIO, deveopers may also register upgrade functions.  In python, each upgrade function takes a dictionary and returns a dictionary.  In C++, the AnyDictionary is manipulated in place.  Each upgrade function is associated with a version number - this is the version number that it upgrades to.
+
+C++ Example (can be viewed/run in `examples/upgrade_downgrade_example.cpp`):
```

Comment:
It could be nice if this was actually linking to the file in the repo. Or maybe we could render this example directly in the documentation?

Reply:
Looks like I might have to do this as a second pass.  Unfortunately the check-links doc steps fails because the file isn't in the expected place... but the example won't exist until the PR is landed.

Reply:
Umm, effectively. At least we know the jobs I added are working!

To avoid this kind of problem, we would have to tell Sphinx that the examples are part of the documentation. Maybe https://linuxtv.org/downloads/sphinx-1.2.3/markup/code.html#directive-literalinclude or https://docutils.sourceforge.io/docs/ref/rst/directives.html#include?

Reply:
Good thought; lets look at this in a later pass.  Thanks!